core/state/statedb.go

```diff
@@ -66,6 +70,34 @@ func (n *proofList) Delete(key []byte) error {
 	panic("not supported")
 }
 
+// CheckTransactionPredicates checks the stateful precompile predicates of any of the given
+// transactions that reference a stateful precompile address in their access list.
+// Returns [len(txs), nil] if and only if all referenced predicates are met.
+// Otherwise, returns the index of the first transaction that was invalid and the predicate error.
+func CheckTransactionPredicates(rules params.Rules, snowCtx *snow.Context, txs types.Transactions, proposerVMBlockCtx *block.Context) (int, error) {
```

Comment:
I don't think statedb is the best place for this. We increase the cross-dependency with params package and it's a cyclic-import hell there. Can we move this to it's own file under direct `core/` folder? 
Do you have another preferred folder in core package? @aaronbuchwald

Reply:
Moved to `core/predicate_check.go`

Reply:
And renamed to `CheckPredicateForSenderTxs` and added doc comments to clarify proper usage.